src/plugins/data/public/index_patterns/fields/field.ts

```diff
@@ -94,7 +94,8 @@ export class Field implements IFieldType {
     if (!type) type = getKbnFieldType('unknown');
 
     let format = spec.format;
-    if (!format || !(format instanceof FieldFormat)) {
```

Comment:
Problem was in this line. We inherit all formatter from the `FieldFormat` but for some reasons `format instanceof FieldFormat === false` =(